src/zenml/materializers/built_in_materializer.py

```diff
@@ -414,6 +414,25 @@ def save(self, data: Any) -> None:
             for entry in metadata:
                 self.artifact_store.rmtree(entry["path"])
             raise e
+        
+    # save dict type objects to JSON file with JSON visualization type
+    def save_visualizations(
+        self, data: Any
+    ) -> Dict[str, "VisualizationType"]:
+        """Save visualizations for the given data.
+
+        Args:
+            data: The data to save visualizations for.
+
+        Returns:
+            A dictionary of visualization URIs and their types.
+        """
+        # dict type objects are always saved as JSON files
+        # doesn't work for non-serializable dict types as they 
+        # are saved as list of lists in different files
+        if isinstance(data, dict) and _is_serializable(data):
```

Comment:
Is this `isinstance(data, dict)` check really necessary? We could also visualize lists/tuples right, they all get stored as a json file anyway if I understand it correctly?